Rename oxalis references to WK

hotzenklotz opened the pull request:
A very simple PR. I renamed to references to Oxalis to WebKnossos:
- `OxalisState` --> `WebknossosState` (store.ts)
- `OxalisApplication` --> `WebKnossosApplication` (app.ts)
- `OxalisApi` --> `WebKnossosApi` (apiHelpers.ts)

The import renaming was automatically handled by VSCode/Typescript language server intergration.

Replaced all occurances of `useSelector((state: OxalisState) => ...)` with  `useWkSelector(state => ...)`.


### Steps to test:
- CI linting and typechecking should be enough

### Issues:
- Contributes to https://github.com/scalableminds/webknossos/issues/8308

------
(Please delete unneeded items, merge only when none are left open)
- [x] Updated [changelog](../blob/master/CHANGELOG.unreleased.md#unreleased)
- [ ] Updated [migration guide](../blob/master/MIGRATIONS.unreleased.md#unreleased) if applicable
- [ ] Updated [documentation](../blob/master/docs) if applicable
- [ ] Adapted [wk-libs python client](https://github.com/scalableminds/webknossos-libs/tree/master/webknossos/webknossos/client) if relevant API parts change
- [ ] Removed dev-only changes like prints and application.conf edits
- [ ] Considered [common edge cases](../blob/master/.github/common_edge_cases.md)
- [ ] Needs datastore update after deployment

philippotto commented:
thank you for the renaming! I think, that the OxalisState --> WebknossosState renaming will potentially cause a lot of merge conflicts. per se this is fine, but if these usages are touched anyway, I'd like to do another change first/too: ensure that `useSelector` is always typed with `WebknossosState`. so, code like this:

```
useSelector((state: OxalisState) => ...)
```

can be turned into 

```
useSelector(state => ...)
```

and TS should still know that state is OxalisState/WkState (similar to how the saga function `select` is typed).

for this typing enhancement, all the `useSelector` usages have to be touched, too. however, doing this first, means that the usages only have to be changed once. if we do the renaming first and then in another pr the type change, we have touched everything twice (= twice the merge conflicts).

can you improve the typing of `useSelector` in this PR? (no need to undo the other change, because it's the same pr).
ideally, we would be able to simply override the type of `useSelector` in react-redux. however, I'm not sure whether this is possible. the alternative would be to create a wrapper à la `useWkSelector`, but then we have to ensure somehow that new PRs don't accidentally reintroduce the usage of `useSelector`. I'm not sure how that could be achieved.

hotzenklotz commented:
I applied the feedback and added a `useWkSelector` shorthand. (see `store.ts`) and replaced all usages of `useSelector((state: OxalisState) => ...)` with that.

philippotto commented on frontend/javascripts/admin/job/job_list_view.tsx:
```diff
@@ -137,7 +137,9 @@ function JobListView() {
   const [jobs, setJobs] = useState<APIJob[]>([]);
   const [searchQuery, setSearchQuery] = useState("");
-  const isCurrentUserSuperUser = useSelector((state: OxalisState) => state.activeUser?.isSuperUser);
+  const isCurrentUserSuperUser = useSelector(
```
useWkSelector?

hotzenklotz replied:
Good catch. I replace another batch of `useSelector` calls.

philippotto replied:
great, however, I think some imports are missing now (see ci output).

philippotto reviewed (approved):
awesome :+1:
